dev/storage.h

```diff
@@ -149,14 +149,32 @@ namespace sqlite_orm {
                         ++suffix;
                     } while(true);
                 }
+                try {
+                    this->start_migration();
 
-                this->create_table(db, backupTableName, tableImpl);
+                    this->create_table(db, backupTableName, tableImpl);
 
-                tableImpl.copy_table(db, backupTableName, columnsToIgnore);
+                    tableImpl.copy_table(db, backupTableName, columnsToIgnore);
 
-                this->drop_table_internal(tableImpl.table.name, db);
+                    this->drop_table_internal(tableImpl.table.name, db);
 
-                tableImpl.rename_table(db, backupTableName, tableImpl.table.name);
+                    tableImpl.rename_table(db, backupTableName, tableImpl.table.name);
+
+                    this->commit_migration();
+
+                } catch(std::exception& ex) {
+                    std::string s = ex.what();  // 's' for debugging
+                    this->abort_migration();
```

Comment:
catching exceptions inside storage functions instead of allowing catching them by users is a bad idea

Reply:
It's rethrown properly just a line below.

Reply:
correct: it is rethown... the reason we have that code inside try/block is that iif the migration fails (as when adding a new column with no default value or not nullablem, then the add_column fails and we need to turn foeign keys ON again!!

Reply:
ok no problem

Reply:
removed!